GliaWidgets/Sources/ViewController/Call/CallViewController.swift

```diff
@@ -4,6 +4,9 @@ final class CallViewController: EngagementViewController {
     private let viewModel: CallViewModel
     private let environment: Environment
     private var proximityManager: ProximityManager?
+    private var hasViewAppeared = false
```

Comment:
It seems like this flag has to be a part of view model, since model is responsible for state.

Reply:
This was moved to `engagementViewController`

Reply:
But this change has same thing - it is still view responsibility, but the idea is to make this flag a part of model.  There's  `viewModel: CommonEngagementModel` in engagementViewController. You can add this property to `CommonEngagementModel`.

Reply:
fixed [53673a7](https://github.com/salemove/ios-sdk-widgets/pull/787/commits/53673a7a4c8426f8042bce05d8385bc7f1db4023)